include/cmm/StructTable.h

```diff
@@ -12,19 +12,50 @@
 
 // Our includes
 #include <cmm/Types.h>
+#include <cmm/Field.h>
 
 // std includes
-#include <optional>
 #include <string>
 #include <unordered_map>
 
 namespace cmm
 {
+    struct StructData
+    {
+        EnumSymState symState;
+        std::unordered_map<std::string, Field> fieldMap;
```

Comment:
Consider using the new IField interface.

Reply:
After reviewing the code, I decided to keep as-is to avoid needing to do extra heap allocation. Any 'getter' type functions will instead use IField, but keep this storage as a "normal" Field.